### PR TITLE
Allow value updates if slidingIndex === -1

### PR DESCRIPTION
**Pull request by adamrneary**
Currently:

1. slidingIndex is initialized as null
1. When a handle is dragged, slidingIndex is set to the index of the
handle being dragged
1. endSlide sets slidingIndex to -1 before removing event listeners and
firing the change event.
1. An onClick handler on the parent div resets slidingIndex to null

The problem is that if the user drags the handle off the parent div
(e.g. resetting to the min value by dragging far to the left), the
onClick handler will not fire, and this leaves the component with
slidingIndex set to -1 indefinitely.

In this state, updateNewValues() will return, so updates to values from
the parent component (e.g. cancelling filters) will not update the
slider.

One solution would be to change onClick to onDragEnd, but drag events
are handled by child elements which call stopPropagation.

This solution simply allows updates if the component is in the state
where slidingIndex === -1.

### CODE REVIEW

**Comment by lencioni** [comment]
I don't have enough context on this code to know if this is the right choice or not, but it would be great to include a test for this change. Will you please add one?

**Comment by adamrneary** [comment]
@lencioni amended. This test is a bit low-level and not particularly semantic, but I am not sure having sliderIndex set to -1 for an indefinite period of time is really a desired state. 

I pinged @goatslacker about the particulars. Happy to adjust as folks see fit.

**Comment by goatslacker** [comment]
I'd rather not have two different states where sliding is allowed. Let's make sure the sliderIndex is reset to `null` always since that is the "no slider index" value.

I believe -1 was added so that we know when we're coming off of a slide so we don't handle the slide's ending twice. There is probably a better way to do this.

**Comment by ritz078** [comment]
What we can do is in case there is a difference in values in props (not state) then set `this.state.slidingIndex` to `null`. I hope this PR gets merged soon.

**Comment by adamrneary** [comment]
@goatslacker i just revised this down to eliminate the `-1` state. PTAL?

File: src/Slider.jsx [hunk 1]
```diff
@@ -428,12 +428,6 @@ export default React.createClass({
 
   // istanbul ignore next
   handleClick(ev) {
-    // if we're coming off of the end of a slide don't handle the click also
```
Comment (goatslacker) [inline src/Slider.jsx]: Is this no longer a problem?

Comment (goatslacker) [inline src/Slider.jsx]: nah I don't think we can do drag testing without writing a lot of code to simulate drag events. I'd rather leave that to the testing library.

Comment (adamrneary) [inline src/Slider.jsx]: Sorry — should have mentioned: I just amended the last commit. It's in there!

File: package.json [hunk 2]
```diff
@@ -21,21 +21,25 @@
...
+    "eslint-config-airbnb": "^11.1.0",
+    "eslint-plugin-import": "^2.0.0",
+    "eslint-plugin-jsx-a11y": "^2.2.2",
+    "eslint-plugin-react": "^6.2.2",
+    "mocha": "^3.0.2",
     "raw-loader": "^0.5.1",
-    "react": "^0.14.3",
-    "react-addons-test-utils": "^0.14.8",
-    "react-dom": "^0.14.3",
+    "react": "^15.3.1",
```
Comment (ljharb) [inline package.json]: we shouldn't update the dev deps here - we're still on 0.14.

Also, updating the dev deps should be done in a separate commit.

Comment (adamrneary) [inline package.json]: I reverted the changes to React deps and isolated the devDependencies bump to a separate commit. Is this what you were looking for? Happy to adjust as needed.
